tests/test_atoms_state.py

```diff
@@ -0,0 +1,386 @@
+        assert orbital_state.degeneracy == 2
+
+
+class TestCoreHole:
+    """
+    Test the `CoreHole` class defined in atoms_state.py.
+    """
+
+    logger = logging.getLogger(__name__)
+
+    @pytest.fixture(autouse=True)
+    def core_hole(self) -> CoreHole:
+        return CoreHole()
+
+    @pytest.mark.parametrize(
+        'orbital_ref, degeneracy, n_excited_electrons, occupation',
+        [
+            (OrbitalsState(l_quantum_number=1), 6, 0.5, 5.5),
+            (OrbitalsState(l_quantum_number=1, ml_quantum_number=-1), 2, 0.5, 1.5),
+            (None, None, 0.5, None),
+        ],
+    )
+    def test_occupation(
+        self, core_hole, orbital_ref, degeneracy, n_excited_electrons, occupation
+    ):
+        """
+        Test the occupation of a core hole for a given set of orbital reference and degeneracy.
+        """
+        core_hole.orbital_ref = orbital_ref
+        if orbital_ref is not None:
+            assert orbital_ref.resolve_degeneracy() == degeneracy
+        core_hole.n_excited_electrons = n_excited_electrons
+        resolved_occupation = core_hole.resolve_occupation(self.logger)
+        if resolved_occupation is not None:
+            assert np.isclose(resolved_occupation, occupation)
+        else:
+            assert resolved_occupation == occupation
+
+    @pytest.mark.parametrize(
+        'orbital_ref, n_excited_electrons, dscf_state, results',
+        [
+            (OrbitalsState(l_quantum_number=1), -0.5, None, (-0.5, None, None)),
+            (OrbitalsState(l_quantum_number=1), 0.5, None, (0.5, None, 5.5)),
+            (
+                OrbitalsState(l_quantum_number=1, ml_quantum_number=-1),
+                0.5,
+                None,
+                (0.5, None, 1.5),
+            ),
+            (OrbitalsState(l_quantum_number=1), 0.5, 'initial', (None, 1, None)),
+            (OrbitalsState(l_quantum_number=1), 0.5, 'final', (0.5, None, 5.5)),
+            (None, 0.5, None, (0.5, None, None)),
+        ],
+    )
+    def test_normalize(
+        self, core_hole, orbital_ref, n_excited_electrons, dscf_state, results
+    ):
+        """
+        Test the normalization of the `CoreHole`. Inputs are defined as the quantities of the `CoreHole` section.
+        """
+        core_hole.orbital_ref = orbital_ref
+        core_hole.n_excited_electrons = n_excited_electrons
+        core_hole.dscf_state = dscf_state
+
+        core_hole.normalize(None, self.logger)
+
+        assert core_hole.n_excited_electrons == results[0]
+        if core_hole.orbital_ref:
+            assert core_hole.orbital_ref.degeneracy == results[1]
+            assert core_hole.orbital_ref.occupation == results[2]
+
+
+class TestHubbardInteractions:
+    """
+    Test the `HubbardInteractions` class defined in atoms_state.py.
+    """
+
+    logger = logging.getLogger(__name__)
+
+    @staticmethod
+    def add_slater_interactions(hubbard_interactions, slater_integrals) -> None:
```

Comment:
this one liner static methods are a bit off for me.

Reply:
Well, I was thinking these could be used to add certain numbers directly to the section passed. To test in other testing units.

Reply:
I deleted the ones that were used once.